File: LAB3/I.py
def bs(a, t, n):
    l, h = 0, n-1
    while(l<=h):
        mid = (l+h)//2
        if a[mid] == t:return mid
        elif a[mid]>t:h = mid - 1
        else:l = mid + 1
    return -1
n = int(input())
a = list(map(int, input().split()))
t = int(input())
if bs(a,t,n) == -1:
    print("No")
else:
    print("Yes")

File: LAB3/I.cpp
#include <iostream>

using namespace std;

int main()
{
    int n; cin >> n;
    int arr[n];
    for(int i = 0 ; i < n; i++){
        cin >> arr[i];
    }
    int target; cin >> target;

    int left = 0, right = n ;

    bool find = false;
    while (left <= right)
    {
        int mid = (left+right)/2;
        if(arr[mid] > target)right = mid - 1;
        else if(arr[mid] < target)left = mid + 1;
        else{
            cout << "Yes";
            find = true;
            break;
        }
    }

    if(find == false)cout << "No";
    
}